mbs_results/constrains.py

```diff
@@ -0,0 +1,170 @@
+import operator
+from typing import List
+
+import pandas as pd
+
+
+def replace_values_index_based(
+    df: pd.DataFrame, target: str, a: int, compare: str, b: int
+) -> None:
+    """
+    Perform comparisons between a subset of df with a and subset of df with b and
+    replace target from a with target from b when comparison is met. Both a
+    and b must exist in the first level index, the comparison is based on the
+    remaining indices.
+
+    Note that this function does not return anything, it modifies the input
+    dataframe.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe to replace the values.
+    target : str
+        Column name for values to be replced.
+    a : int
+        Question_no to check.
+    compare : str
+        Logical operator to compare, accepted '>', '<','>=','<=','=='.
+    b : int
+        Question_no to check against.
+    """
+    # For improved perfomance
+    df.sort_index(inplace=True)
+
+    ops = {
+        ">": operator.gt,
+        "<": operator.lt,
+        ">=": operator.ge,
+        "<=": operator.le,
+        "==": operator.eq,
+    }
+
+    series_from_a = df.loc[a][target]
+
+    series_from_b = df.loc[b][target]
+
+    common_index = series_from_a.index.intersection(series_from_b.index)
+
+    # Has format (period,reference)
+    index_to_replace = series_from_a[
+        ops[compare](series_from_a[common_index], series_from_b[common_index])
+    ].index
+
+    if len(index_to_replace) > 0:
+        for date_ref_idx in index_to_replace.values:
+
+            # Has format (question,no,period,reference)
```

Comment:
is the format not `question_no, period, reference`?

Reply:
yep you are right